.test.mk

```diff
@@ -0,0 +1,201 @@
+#
+# CI testing rules
+#
+
+SRC_DIR = .
+BUILD_DIR ?= .
+VARDIR ?= /tmp/t
+MAX_FILES ?= 4096
+
+LUAJIT_TEST_BUILD_DIR ?= ${BUILD_DIR}
+TEST_RUN_PARAMS ?= --builddir ${PWD}/${BUILD_DIR}
+
+configure:
+	${CMAKE_ENV} cmake -S ${SRC_DIR} -B ${BUILD_DIR} ${CMAKE_PARAMS} ${CMAKE_EXTRA_PARAMS}
+
+# Static Analysis
+
+luacheck: configure
+	${CMAKE_ENV} cmake --build ${BUILD_DIR} --parallel $(NPROC) --target luacheck
+
+# Building
+
+NPROC=$(shell nproc || sysctl -n hw.ncpu)
+
+build: configure
+	${CMAKE_ENV} cmake --build ${BUILD_DIR} --parallel $(NPROC)
+	if [ "${CTEST}" = "true" ]; then cd ${BUILD_DIR} && ctest -V; fi
+
+# Testing
+
+run-luajit-test:
+	${CMAKE_ENV} cmake --build ${LUAJIT_TEST_BUILD_DIR} --parallel $(NPROC) --target LuaJIT-test
+
+install-test-deps:
+	python3 -m pip install -r test-run/requirements.txt
+
+run-test: install-test-deps
+	cd test && ${TEST_RUN_ENV} ./test-run.py --force --vardir ${VARDIR} ${TEST_RUN_PARAMS} ${TEST_RUN_EXTRA_PARAMS}
+
+##############################
+# Linux                      #
+##############################
+
+# Release build
+
+test-release: CMAKE_PARAMS = -DCMAKE_BUILD_TYPE=RelWithDebInfo -DENABLE_WERROR=ON
+test-release: build run-luajit-test run-test
+
+# Release ASAN build
+
+test-release-asan: CMAKE_ENV = CC=clang-11 CXX=clang++-11
+test-release-asan: CMAKE_PARAMS = -DCMAKE_BUILD_TYPE=RelWithDebInfo -DENABLE_WERROR=ON -DENABLE_ASAN=ON \
+	-DENABLE_UB_SANITIZER=ON -DENABLE_FUZZER=ON
+# Temporary suppressed some checks in the scope of the issue https://github.com/tarantool/tarantool/issues/4360:
+#  - ASAN: to suppress failures of memory error checks caught while tests run, the tarantool/asan/asan.supp file is
+#      used. It is set as a value for -fsanitize-blacklist option at the build time in the cmake/profile.cmake file.
+#  - LSAN: to suppress failures of memory leak checks caught while tests run, the tarantool/asan/lsan.supp file is used.
+test-release-asan: TEST_RUN_ENV = ASAN=ON LSAN_OPTIONS=suppressions=${PWD}/asan/lsan.supp \
+	ASAN_OPTIONS=heap_profile=0:unmap_shadow_on_exit=1:detect_invalid_pointer_pairs=1:symbolize=1:detect_leaks=1:dump_instruction_bytes=1:print_suppressions=0
```

Comment:
Fixed.

Reply:
I am using 120 symbols per line. IMHO, it is ok.

Reply:
Okay, fixed. `80 symbols per line` rule is used.

Reply:
Eventually, it was decided in a private discussion not to use 80 symbols limitation because in this case we have to break lines at many places and it looks ugly. But some places were formatted by @ligurio's suggestions and now they look nice.